### PR TITLE
Purchases: Rename constants for upgrade actions to be more consistent

### PR DESCRIPTION
**Pull request by stephanethomas**
This pull request makes sure that constants for store events all follow the same naming convention. This naming convention - which is heavily inspired from [Ian Obermiller's post](https://news.ycombinator.com/item?id=7721381) - was discussed some times ago internally but we never found the time to actually update our code. Well, I had some free time during my flight from London to Philadelphia so here it goes :).

The logic didn't change and everything should work as before.

### Review
* [x] Code review
* [x] QA review

### CODE REVIEW

File: client/lib/upgrades/constants.js [hunk 1]
```diff
@@ -16,40 +25,26 @@ module.exports.action = keyMirror( {
...
 	PRIMARY_DOMAIN_SET: null,
 	PRIMARY_DOMAIN_SET_COMPLETED: null,
 	PRIMARY_DOMAIN_SET_FAILED: null,
-	PURCHASES_PRIVATE_REGISTRATION_CANCEL: null,
-	PURCHASES_PRIVATE_REGISTRATION_CANCEL_COMPLETED: null,
-	PURCHASES_PRIVATE_REGISTRATION_CANCEL_FAILED: null,
+	PRIVACY_PROTECTION_CANCEL: null,
+	PRIVACY_PROTECTION_CANCEL_COMPLETED: null,
+	PRIVACY_PROTECTION_CANCEL_FAILED: null,
+	PRIVACY_PROTECTION_ENABLE_COMPLETED: null,
```
Comment (gziolo) [inline client/lib/upgrades/constants.js]: This is one looks like a lonely one ;) Of course not related to this PR, but we should double check that part of Domain Management.

**Comment by gziolo** [comment]
Code changes look awesome. Great job on this one :+1: 
I haven't testes this one so leaving in `Needs review` state.

File: client/lib/upgrades/constants.js (hunk 1 above)
Comment (stephanethomas) [inline client/lib/upgrades/constants.js]: There are in fact other inconsistencies. For example the delete DNS action is plain wrong: [we're currently missing a `DNS_DELETE_COMPLETED` event](https://github.com/Automattic/wp-calypso/blob/de5df67d8c0f27782e9e379af6f86f8bfc30c205/client/lib/upgrades/actions/domain-management.js#L230) and [we're expecting an `error` object in the payload of the `DELETING_DNS` event](https://github.com/Automattic/wp-calypso/blob/de5df67d8c0f27782e9e379af6f86f8bfc30c205/client/lib/domains/dns/reducer.js#L48) that we never pass. And I'm not talking of how we handle errors :).

@fabianapsimoes, we should probably take care of that at some point.

**Comment by scruffian** [comment]
I tested domain management, domain search, plans, purchases, checkout. If there's nothing else to test this LGTM :+1:

File: client/lib/upgrades/constants.js (hunk 1 above)
Comment (fabianapsimoes) [inline client/lib/upgrades/constants.js]: @stephanethomas can you open a new issue for that? I'm guessing this falls under Cobalt's realm, so pinging @aidvu.

Comment (aidvu) [inline client/lib/upgrades/constants.js]: Thanks @fabianapsimoes. @klimeryk will pick it probably.

**Pull request merged**